### PR TITLE
Docker: signal forwarding & migrations in entrypoint

### PR DESCRIPTION
**Pull request by marcklingen**
Changes

* signal forwarding using dumb-init

* prisma migrations applied in entrypoint

Fixes

* Add SALT to args of docker-compose.build.yml

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -79,11 +82,12 @@ COPY --from=builder --chown=nextjs:nodejs /app/prisma ./prisma
...
+RUN chmod +x ./entrypoint.sh
 
 USER nextjs
 
 # Default port to 3000
 ENV PORT 3000
 
 # CMD ["node", "server.js"]
-CMD ["/bin/sh", "entrypoint.sh"]
+CMD ["dumb-init", "--", "./entrypoint.sh"]
```
Comment (maxdeichmann) [inline Dockerfile:93]: Was macht `dumb-init`?

Comment (maxdeichmann) [reply Dockerfile:93]: I think best practice would be to have the migration on the docker-compose level as it is more obvious to an engineer that this migration step is happening with each docker container start. However, i understand it is easier here as we have access to prisma here. Hence i think it is ok.

File: README.md [hunk 2]
```diff
@@ -107,9 +107,6 @@ cd langfuse
 
 # Run server and database
 docker compose up -d
-
-# Apply db migrations
-DATABASE_URL=postgresql://postgres:postgres@localhost:5432/postgres DIRECT_URL=postgresql://postgres:postgres@localhost:5432/postgres npx prisma migrate deploy
```
Comment (maxdeichmann) [inline README.md:112]: We also need to adjust the docs then i assume

Comment (marcklingen) [reply README.md:112]: Yes. Will do once we merge.

File: Dockerfile (hunk 1 above)
Comment (marcklingen) [reply Dockerfile:93]: See description: signal forwarding using dumb-init
- eg ctrl-c or shutdown requests. Nodejs doesn't handle them well when run as PID 1

Comment (marcklingen) [reply Dockerfile:93]: Goal: migrations work as part of the startup of the container wie package. Needs to also work in the non-docker-compose workflow when deploying the container

**Pull request merged**